### PR TITLE
CoreFX CI Unix Staging commit

### PR DESCRIPTION
**Pull request by A-And**
Modifying NETCI definition to stage for Unix version of  #18365

### CODE REVIEW

**Comment by A-And** [comment]
@dotnet-bot test ci please

**Comment by A-And** [comment]
@dotnet-bot test ci please

**Comment by A-And** [comment]
@dotnet-bot test ci please

**Comment by A-And** [comment]
@dotnet-bot test ci please

**Comment by A-And** [comment]
@BruceForstall - Sure thing! I had misunderstood that we'd disable Release configurations entirely.

**Comment by A-And** [comment]
@dotnet-bot test ci please

**Comment by A-And** [comment]
@dotnet-bot test ci please

**Comment by A-And** [comment]
@dotnet-bot test ci please

**Comment by A-And** [comment]
@dotnet-bot test ci please

**Comment by A-And** [comment]
@dotnet-bot test ci please

**Comment by A-And** [comment]
@dotnet-bot test ci please

**Comment by A-And** [comment]
@dotnet-bot test ci please

**Comment by A-And** [comment]
@dotnet-bot test ci please

File: netci.groovy [hunk 1]
```diff
@@ -2808,10 +2840,10 @@ def static shouldGenerateJob(def scenario, def isPR, def architecture, def confi
                 }
                 break
             case 'corefx_innerloop':
-                if (os != 'Windows_NT'|| architecture != 'x64') {
+                if ( (os != 'Windows_NT' && os != 'Ubuntu' &&  os != 'OSX10.12') ||  architecture != 'x64') {
```
Comment (BruceForstall) [inline netci.groovy]: Can you please split this into two conditions:
```
if (os != 'Windows_NT' && os != 'Ubuntu' &&  os != 'OSX10.12') {
   return false
}
if (architecture != 'x64') {
  return false
}
```

File: netci.groovy [hunk 2]
```diff
@@ -2808,10 +2840,10 @@ def static shouldGenerateJob(def scenario, def isPR, def architecture, def confi
...
                     return false
                 }
-                if(configuration != 'Release' && configuration != 'Checked') {
+                if(configuration != 'Checked') {
```
Comment (BruceForstall) [inline netci.groovy]: I don't think there's any reason to restrict this to just Checked. Seems like we should just delete this condition and allow all of Debug/Checked/Released jobs.

File: netci.groovy [hunk 3]
```diff
@@ -1565,6 +1565,14 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
                         Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${architecture} ${configuration} via ILLink", "(?i).*test\\W+${os}\\W+${architecture}\\W+${configuration}\\W+${scenario}.*")
                         break
                     }
+
+                    else  if (scenario == 'corefx_innerloop') {
+                        if (configuration == 'Checked') {
```
Comment (BruceForstall) [inline netci.groovy]: To allow non-debug triggers for Debug/Release, this (and the equivalent other code, below) should be:
```
if (configuration == 'Checked') {
  Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${architecture} ${configuration} CoreFX Tests")
} else {
  Utilities.addGithubPRTriggerForBranch(job, branch, "${os} ${architecture} ${configuration} CoreFX Tests", "(?i).*test\\W+${os}\\W+${architecture}\\W+${configuration}\\W+CoreFX Tests.*")
}
```

**Comment by A-And** [comment]
@dotnet-bot test ci please

**Comment by A-And** [comment]
@dotnet-bot test ci please

**Comment by A-And** [comment]
@dotnet-bot test ci please

**Pull request merged**